File: src/main/java/nuclearcoder/discordbot/command/misc/Cmd8ball.java
package nuclearcoder.discordbot.command.misc;

import nuclearcoder.discordbot.NuclearBot;
import nuclearcoder.discordbot.command.Command;
import sx.blah.discord.handle.obj.IChannel;
import sx.blah.discord.handle.obj.IMessage;
import sx.blah.discord.handle.obj.IUser;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class Cmd8ball implements Command {

    private String[] answers = new String[20];
    private Random random = new Random();

    {
        List<String> answerList = new ArrayList<>(answers.length);

        // yes
        answerList.add("It is certain");
        answerList.add("It is decidedly so");
        answerList.add("Without a doubt");
        answerList.add("Yes, definitely");
        answerList.add("You may rely on it");
        answerList.add("As I see it, yes");
        answerList.add("Most likely");
        answerList.add("Outlook good");
        answerList.add("Yes");
        answerList.add("Signs point to yes");

        // maybe
        answerList.add("Reply hazy try again");
        answerList.add("Ask again later");
        answerList.add("Better not tell you now");
        answerList.add("Cannot predict now");
        answerList.add("Concentrate and ask again");

        // no
        answerList.add("Don't count on it");
        answerList.add("My reply is no");
        answerList.add("My sources say no");
        answerList.add("Outlook not so good");
        answerList.add("Very doubtful");

        answerList.toArray(answers);
    }

    @Override public void execute(NuclearBot bot, IMessage message, String command, String[] args)
    {
        IChannel channel = message.getChannel();
        IUser user = message.getAuthor();

        String answer = answers[random.nextInt(answers.length)];

        bot.sendMessage(channel, user.mention() + "\n:8ball: " + answer);
    }

}

File: src/main/java/nukebot/command/misc/Cmd8ball.kt
package nukebot.command.misc

import nukebot.NuclearBot
import nukebot.command.Command
import nukebot.rSendMessage
import sx.blah.discord.handle.obj.IMessage
import java.util.*

class Cmd8ball : Command {

    private val random = Random()
    private val answers = arrayOf(
            // yes
            "It is certain",
            "It is decidedly so",
            "Without a doubt",
            "Yes, definitely",
            "You may rely on it",
            "As I see it, yes",
            "Most likely",
            "Outlook good",
            "Yes",
            "Signs point to yes",

            // maybe
            "Reply hazy try again",
            "Ask again later",
            "Better not tell you now",
            "Cannot predict now",
            "Concentrate and ask again",

            // no
            "Don't count on it",
            "My reply is no",
            "My sources say no",
            "Outlook not so good",
            "Very doubtful"
    )

    override fun execute(bot: NuclearBot, message: IMessage, command: String, args: Array<String>) {
        val channel = message.channel
        val user = message.author

        val answer = answers[random.nextInt(answers.size)]

        channel.rSendMessage(user.mention() + "\n:8ball: " + answer)
    }

}
